custom_components/elasticsearch/es_gateway.py

```diff
@@ -42,644 +48,169 @@ class ElasticsearchGateway(ABC):
-
-        return args
-
-    async def _get_cluster_info(self) -> dict:
-        """Retrieve info about the connected elasticsearch cluster."""
-        try:
-            info = await self._client.info()
-        except Exception:
-            self.convert_es_error()
-
-        if not isinstance(info, dict):
-            msg = "Invalid response from Elasticsearch"
-            self.convert_es_error(msg)
-
-        return dict(info)
+        minimum_privileges: MappingProxyType[str, Any] = ES_CHECK_PERMISSIONS_DATASTREAM,
+        log: Logger = BASE_LOGGER,
+    ) -> None:
+        """Initialize the gateway and then stop it."""
 
     @abstractmethod
-    async def bulk(self, **kwargs) -> None:
-        """Perform a bulk operation."""
-
-    def _convert_api_response_to_dict(self, response: object) -> dict:
-        """Convert an API response to a dictionary."""
-        if not isinstance(response, dict):
-            msg = "Invalid response from Elasticsearch"
-            raise TypeError(msg)
+    async def info(self) -> dict:
+        """Retrieve info about the connected elasticsearch cluster."""
 
-        return dict(response)
+    async def check_connection(self) -> bool:
+        """Check if the connection to the Elasticsearch cluster is working."""
 
-    async def get_index_template(self, **kwargs) -> dict:
-        """Retrieve an index template."""
-        self._logger.debug("Retrieving index template %s", kwargs.get("name", ""))
+        previous_ping = self._previous_ping
+        new_ping = await self.ping()
 
-        try:
-            result = await self._client.indices.get_template(**kwargs)
-
-        except Exception:
-            msg = "Error retrieving index template"
-            self.convert_es_error(msg)
+        # Our first connection check
+        if previous_ping is None:
+            established = new_ping
+            if established:
+                self._logger.info("Connection to Elasticsearch is established.")
+            else:
+                self._logger.error("Failed to establish connection to Elasticsearch.")
 
-        if not isinstance(result, dict):
-            self._logger.error("Invalid response from Elasticsearch")
+            return new_ping
 
-        return self._convert_api_response_to_dict(result)
+        reestablished: bool = not previous_ping and new_ping
+        maintained = previous_ping and new_ping
+        lost: bool = previous_ping and not new_ping
+        down: bool = not previous_ping and not new_ping
 
-    async def put_index_template(self, **kwargs) -> dict:
-        """Retrieve an index template."""
-        self._logger.debug("Updating index template %s", kwargs.get("name", ""))
+        if maintained:
+            self._logger.debug("Connection to Elasticsearch is still available.")
 
-        try:
-            result = await self._client.indices.put_index_template(**kwargs)
+        if lost:
+            self._logger.error("Connection to Elasticsearch has been lost.")
```

Comment:
Without debug logging, this is the only message I receive when an API Key has insufficnent credentials. This is a bit misleading as it leads me to believe that there is a connection issue, as opposed to an authorization issue. That being said, the integration sucessfully reloads and prompts the user for new credentials, which is great!

A more descriptive log message would be nice to have, but I don't consider this a blocker.

Reply:
Will look at this

Reply:
Should be fixed.

I also added the root cause of the auth issue as reported by Elasticsearch into the message in the UI.